### PR TITLE
numworks-epsilon: 15.5.0 -> 22.2.0

### PR DESCRIPTION
**Pull request by imadnyc**
## Description of changes

Updated package
## Things done
Added python lz4. Tested new features since last release and it works. 

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/science/math/numworks-epsilon/default.nix [hunk 1]
```diff
@@ -9,17 +9,18 @@
...
   pname = "numworks-epsilon";
-  version = "15.5.0";
+  version = "22.2.0";
 
   src = fetchFromGitHub {
     owner = "numworks";
     repo = "epsilon";
     rev = version;
-    sha256 = "fPBO3FzZ4k5OxG+Ifc6R/au4Te974HNKAEdHz+aFdSg=";
+    sha256 = "sha256-E2WaXTn8+Ky9kdZxvQmEt63Ggo6Ns0fZ0Za+rQGIMSg=";
```
Comment (jtbx) [inline pkgs/applications/science/math/numworks-epsilon/default.nix:23]: `hash` should be used wherever possible
```suggestion
    hash = "sha256-E2WaXTn8+Ky9kdZxvQmEt63Ggo6Ns0fZ0Za+rQGIMSg=";
```

File: pkgs/applications/science/math/numworks-epsilon/default.nix [hunk 2]
```diff
@@ -31,17 +32,18 @@ stdenv.mkDerivation rec {
...
-  patches = [
-    # Remove make rule Introduced in cba596dde7
-    # which causes it to not build with nix
-    ./0001-ion-linux-makerules.patch
-  ];
+  # patches = [
+  #   # Remove make rule Introduced in cba596dde7
+  #   # which causes it to not build with nix
+  #   ./0001-ion-linux-makerules.patch
+  # ];
```
Comment (jtbx) [inline pkgs/applications/science/math/numworks-epsilon/default.nix:46]: Why is this commented out? If it doesn't matter you can just remove it. The upstream file is already patched.

**Review by jtbx — CHANGES_REQUESTED** [review]

File: pkgs/applications/science/math/numworks-epsilon/default.nix (hunk 2 above)
Comment (imadnyc) [reply pkgs/applications/science/math/numworks-epsilon/default.nix]: You're right, forgot to remove it

**Review by jtbx — APPROVED** [review]
Looks good :ok_hand:

File: pkgs/applications/science/math/numworks-epsilon/default.nix [hunk 3]
```diff
@@ -31,18 +32,13 @@ stdenv.mkDerivation rec {
     python3
     imagemagick
     gcc-arm-embedded
+    python311Packages.lz4
```
Comment (pbsds) [inline pkgs/applications/science/math/numworks-epsilon/default.nix:35]:
```suggestion
    python3.pkgs.lz4
```

Comment (pbsds) [reply pkgs/applications/science/math/numworks-epsilon/default.nix:35]: or
```suggestion
    python3Packages.lz4
```

Comment (imadnyc) [reply pkgs/applications/science/math/numworks-epsilon/default.nix]: Done

**Pull request merged**